src/taipy/core/submission/submission.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import uuid
+from datetime import datetime
+from typing import List, Optional, Union
+
+from .._entity._entity import _Entity
+from .._entity._labeled import _Labeled
+from .._entity._reload import _self_reload, _self_setter
+from .._version._version_manager_factory import _VersionManagerFactory
+from ..job._job_manager_factory import _JobManagerFactory
+from ..job.job import Job, JobId
+from .submission_id import SubmissionId
+from .submission_status import SubmissionStatus
+
+
+class Submission(_Entity, _Labeled):
+    """Hold the jobs and submission status when a Scenario^, Sequence^ or Task^ is submitted.
+
+    Attributes:
+        entity_id (str): The identifier of the entity that was submitted.
+        id (str): The identifier of the `Submission^` entity.
+        jobs (Optional[Union[List[Job], List[JobId]]]): A list of jobs.
+        creation_date (Optional[datetime]): The date of this submission's creation.
+        submission_status (Optional[SubmissionStatus]): The current status of this submission.
+        version (Optional[str]): The string indicates the application version of the submission to instantiate.
+            If not provided, the latest version is used.
+    """
+
+    _ID_PREFIX = "SUBMISSION"
+    _MANAGER_NAME = "submission"
+    __SEPARATOR = "_"
+
+    def __init__(
+        self,
+        entity_id: str,
+        id: Optional[str] = None,
+        jobs: Optional[Union[List[Job], List[JobId]]] = None,
```

Comment:
❓Just by curiosity, do we really need to have `Union[List[Job], List[JobId]]`? Why not a simpler `List[JobId]`? 
Having a more precise parameter type lower the risk of bugs in the long term.
M2c

Reply:
I want to offer both possibility to pass in the Job entity or just the JobId instead of asking the user for 1 of the 2 if that makes sense to you :D since we use entity and id interchangeably, enforcing just one might be going away from the overall behavior. Though it probably still makes sense in this context as Submission is like for internal use only (at least for now)

Reply:
I leave you the final decision of course, and it's not a big deal anyway :-)

But as a user, I think I would prefer clear and explicit API. 

With the one you propose, the caller may wonder if the function work differently. For example, would the function avoid any additional data loading if we pass the job?
Again, not really a pb, and we can deal with this API :-)

Reply:
If we want to change this convention, or at least discuss it, it should be a global decision. It is out of this PR scope.

Reply:
I agree :D as the behavior is quite global, it's not the best place to discuss it!